lightning/src/ln/onion_utils.rs

```diff
@@ -2504,40 +2750,147 @@ mod tests {
 	}
 
 	#[test]
-	fn test_failure_packet_onion() {
-		// Returning Errors test vectors from BOLT 4
+	fn test_attributable_failure_packet_onion_mutations() {
+		// Define the length of the (legacy) failure message field in the test.
+		const FAILURE_MESSAGE_LEN: usize = 1060;
+
+		for mutating_node in 0..5 {
+			for mutated_index in
+				0..FAILURE_MESSAGE_LEN + HOLD_TIME_LEN * MAX_HOPS + HMAC_LEN * HMAC_COUNT
+			{
+				let mutation = Mutation { node: mutating_node, index: mutated_index };
+				let decrypted_failure =
+					test_attributable_failure_packet_onion_with_mutation(Some(mutation));
+
+				if decrypted_failure.onion_error_code == Some(16399) {
```

Comment:
nitnit: `0x4000 | 10` for the failure code like it's done elsewhere in the codebase